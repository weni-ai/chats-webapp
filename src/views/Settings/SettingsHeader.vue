<template>
  <header class="settings-view__header">
    <UnnnicAvatarIcon
      icon="settings"
      scheme="weni-600"
    />
    <h1 class="header__title">{{ $t('config_chats.title') }}</h1>
  </header>
</template>
<script>
export default {
  name: 'SettingsHeader',
};
</script>

<style lang="scss" scoped>
.settings-view__header {
  display: flex;
  align-items: center;
  gap: $unnnic-spacing-xs;

  .header__title {
    color: $unnnic-color-neutral-darkest;
    font-family: $unnnic-font-family-primary;
    font-size: $unnnic-font-size-title-sm;
    font-weight: $unnnic-font-weight-bold;
    line-height: $unnnic-font-size-title-sm + $unnnic-line-height-md;
  }
}
</style>
