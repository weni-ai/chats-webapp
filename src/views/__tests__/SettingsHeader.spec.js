import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';

import SettingsHeader from '@/views/Settings/SettingsHeader.vue';

const createWrapper = (props) => {
  return mount(SettingsHeader, {
    props,
    global: {
      stubs: {
        UnnnicAvatarIcon: true,
      },
    },
  });
};

describe('SettingsHeader.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('should render the UnnnicAvatarIcon with correct props', () => {
    const avatarIcon = wrapper.findComponent(
      '[data-testid=settings-view-header-icon]',
    );

    expect(avatarIcon.exists()).toBe(true);
    expect(avatarIcon.props('icon')).toBe('settings');
    expect(avatarIcon.props('scheme')).toBe('weni-600');
  });

  it('should render the header title with correct text', () => {
    const headerTitle = wrapper.find(
      '[data-testid=settings-view-header-title]',
    );

    expect(headerTitle.exists()).toBe(true);
    expect(headerTitle.text()).toBe(wrapper.vm.$t('config_chats.title'));
  });

  it('Should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
