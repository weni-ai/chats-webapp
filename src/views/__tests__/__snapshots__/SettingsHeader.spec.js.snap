// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsHeader.vue > Should match the snapshot 1`] = `
<header
  class="settings-view__header"
  data-v-a59c663f=""
>
  <div
    class="unnnic-avatar-icon weni-600 sm"
    data-testid="settings-view-header-icon"
    data-v-a59c663f=""
    data-v-ecaaa6f3=""
  >
    <span
      class="material-symbols-rounded unnnic-icon-scheme--weni-600 unnnic-icon-size--md"
      data-testid="material-icon"
      data-v-76ceda5d=""
      data-v-ecaaa6f3=""
      translate="no"
    >
      settings
    </span>
  </div>
  <h1
    class="header__title"
    data-testid="settings-view-header-title"
    data-v-a59c663f=""
  >
    Chat Settings
  </h1>
</header>
`;
